tests/test_muon_preproc.py

```diff
@@ -132,6 +145,16 @@ def test_filter_var_adata_view(self, mdata, filepath_h5mu):
             sub = np.random.binomial(1, 0.5, view.n_vars).astype(bool)
             mu.pp.filter_var(view, sub)
 
+    def test_filter_var_with_varm_varp(self, pbmc3k_processed):
+        A = pbmc3k_processed[:500,].copy()
+        B = pbmc3k_processed[500:,].copy()
+        mdata = mu.MuData({"A": A, "B": B})
+        np.random.seed(42)
+        var_sel = np.random.choice(np.array([0, 1]), size=mdata.n_vars, replace=True)
+        mdata.var["sel"] = var_sel
+        mu.pp.filter_var(mdata, "sel", lambda y: y == 1)
+        assert mdata.shape[1] == int(np.sum(var_sel))
```

Comment:
Update this test as well to check the full object like `     assert_equal(mdata["B"], B_subset)`